src/otaclient/app/ota_client.py

```diff
@@ -593,33 +645,34 @@ def _on_failure(self, exc: ota_errors.OTAError, ota_status: api_types.StatusOta)
                 exc.get_error_report(f"OTA failed with {ota_status.name}: {exc!r}")
             )
         finally:
-            exc = None  # type: ignore , prevent ref cycle
+            del exc  # prevent ref cycle
 
     # API
 
-    def update(self, version: str, url_base: str, cookies_json: str):
+    def update(self, version: str, url_base: str, cookies_json: str) -> None:
         if self._lock.acquire(blocking=False):
             try:
-                logger.info("[update] entering...")
+                logger.info("[update] entering local update...")
                 self._update_executor = _OTAUpdater(
+                    version,
+                    url_base,
+                    cookies_json,
                     boot_controller=self.boot_controller,
                     create_standby_cls=self.create_standby_cls,
                     control_flags=self.control_flags,
-                    proxy=self.proxy,
+                    upper_otaproxy=self.proxy,
                 )
 
-                # reset failure information on handling new update request
                 self.last_failure_type = api_types.FailureType.NO_FAILURE
                 self.last_failure_reason = ""
                 self.last_failure_traceback = ""
 
-                # enter update
                 self.live_ota_status.set_ota_status(api_types.StatusOta.UPDATING)
-                self._update_executor.execute(version, url_base, cookies_json)
+                self._update_executor.execute()
             except ota_errors.OTAError as e:
                 self._on_failure(e, api_types.StatusOta.FAILURE)
             finally:
-                self._update_executor = None  # type: ignore
+                self._update_executor = None
                 gc.collect()  # trigger a forced gc
                 self._lock.release()
```

Comment:
OTAUpdater instance is OTA session specific, for each OTA update request we will create a new instance of OTAUpdater.